File: BackEnd/src/main/java/com/bezkoder/springjwt/models/metier/Enrollment.java
package com.bezkoder.springjwt.models.metier;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Enrollment {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne
    private Student student;
    @OneToOne
    private Course course;
    private LocalDate enrollmentDate;
}

File: FrontEnd/src/app/model/enrollment.ts
import { Course } from './course';

export class Enrollment {
  id: number = 0;
  course?: Course;
  enrollmentDate?: Date;

  constructor(id: number, course: Course, enrollmentDate: Date) {
    this.id = id;
    this.course = course;
    this.enrollmentDate = enrollmentDate;
  }
}
